src/test/scala/io.circe.config/CirceConfigSpec.scala

```diff
@@ -70,6 +72,18 @@ class CirceConfigSpec extends FlatSpec with Matchers {
     assert(AppConfig.as[Nested]("e") == Right(Nested(true)))
   }
 
+  type ErrorHandler[A] = Either[Throwable, A] // Typically this would be IO or something similar
```

Comment:
I suggest to bring in `cats-effect` as a test dependency so we can use it in the tests below.

Reply:
Added cats effect and more tests in 6d76b92